test/sample-test/sample_test_launcher.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This launcher module serves as the entry-point of the sample test image. It
+decides which test to trigger based upon the arguments provided.
+"""
+
+import fire
+import os
+import papermill as pm
+import subprocess
+import uuid
+
+from google.cloud import storage
+
+
+# List of notebook samples' test names and corresponding file names.
+NOTEBOOK_SAMPLES = {
+    'kubeflow_pipeline_using_TFX_OSS_components': 'KubeFlow Pipeline Using TFX OSS Components.ipynb',
+    'lightweight_component': 'Lightweight Python components - basics.ipynb',
+    'dsl_static_type_checking': 'DSL Static Type Checking.ipynb'
+}
+
+
+PROJECT_NAME = 'ml-pipeline-test'
+PAPERMILL_ERR_MSG = 'An Exception was encountered at'
+
+
+#TODO(numerology): Add unit-test for classes.
+class SampleTest(object):
+  """Launch a KFP sample_test provided its name.
+
+  Args:
+    test_name: name of the sample test.
+    input: The path of a pipeline package that will be submitted.
+    result: The path of the test result that will be exported.
+    output: The path of the test output.
+    namespace: Namespace of the deployed pipeline system. Default: kubeflow
+  """
+
+  GITHUB_REPO = 'kubeflow/pipelines'
+  BASE_DIR= '/python/src/github.com/' + GITHUB_REPO
+  TEST_DIR = BASE_DIR + '/test/sample-test'
+
+  def __init__(self, test_name, test_results_gcs_dir, target_image_prefix,
+                         namespace='kubeflow'):
+    self._test_name = test_name
+    self._results_gcs_dir = test_results_gcs_dir
+    #(TODO: numerology) target_image_prefix seems to be only used for post-submit
+    # check.
+    self._target_image_prefix = target_image_prefix
+    self._namespace = namespace
+    self._sample_test_result = 'junit_Sample%sOutput.xml' % self._test_name
+    self._sample_test_output = self._results_gcs_dir
+    self._work_dir = self.BASE_DIR + '/samples/core/' + self._test_name
+
+    self._run_test()
+
+  def check_result(self):
+    os.chdir(self.TEST_DIR)
+    subprocess.call([
+        'python3',
```

Comment:
Nit: `sys.executable` is a bit better since it points to the working python binary.

Reply:
Thanks! Will do.